go-controller/pkg/ovn/base_network_controller_policy.go

```diff
@@ -519,106 +501,124 @@ func getPolicyType(policy *knet.NetworkPolicy) (bool, bool) {
 
 // getNewLocalPolicyPorts will find and return port info for every given pod obj, that is not found in
 // np.localPods.
-// if there are problems with fetching port info from logicalPortCache, pod will be added to errObjs.
-func (oc *DefaultNetworkController) getNewLocalPolicyPorts(np *networkPolicy,
-	objs ...interface{}) (policyPortsToUUIDs map[string]string, policyPortUUIDs []string, errObjs []interface{}) {
+// if there are problems with fetching port info from logicalPortCache, error will be added to returned error array.
+func (bnc *BaseNetworkController) getNewLocalPolicyPorts(np *networkPolicy,
+	objs ...interface{}) (policyPortsToUUIDs map[string]string, policyPortUUIDs []string, errs []error) {
 
-	policyPortUUIDs = make([]string, 0, len(objs))
+	policyPortUUIDs = []string{}
 	policyPortsToUUIDs = map[string]string{}
+	errs = []error{}
 
 	for _, obj := range objs {
 		pod := obj.(*kapi.Pod)
-
-		logicalPortName := util.GetLogicalPortName(pod.Namespace, pod.Name)
-		if _, ok := np.localPods.Load(logicalPortName); ok {
-			// port is already added for this policy
-			continue
-		}
-
 		if pod.Spec.NodeName == "" {
 			// pod is not yet scheduled, will receive update event for it
 			continue
 		}
 
 		// Skip pods that will never be present in logicalPortCache,
 		// e.g. hostNetwork pods, overlay node pods, or completed pods
-		if !oc.podExpectedInLogicalCache(pod) {
+		if !bnc.podExpectedInLogicalCache(pod) {
 			continue
 		}
 
-		// Add pod to errObjs for retry if
-		// 1. getting pod LSP from the cache fails,
-		// 2. the gotten LSP is scheduled for removal (stateful-sets).
-		portInfo, err := oc.logicalPortCache.get(pod, types.DefaultNetworkName)
-		if err != nil {
-			klog.Warningf("Failed to get LSP for pod %s/%s for networkPolicy %s, err: %v",
-				pod.Namespace, pod.Name, np.name, err)
-			errObjs = append(errObjs, pod)
-			continue
-		}
+		nadNames := bnc.getPodNADNames(pod)
+		for _, nadName := range nadNames {
+			var logicalPortName string
```

Comment:
`logicalPortName := bnc.GetLogicalPortName(pod, nadName)` ?

Reply:
are you suggesting the change the function signature of GetSecondaryNetworkLogicalPortName(), or you meant to have one single function for both default and secondary network?

for latter, the problem is that we'd still need to check if it is default network, to decide weather or not prefix the logical port name. I believe that is the reason why we agreed to have two different functions.

Reply:
I mean this function already exists
```
func (bnc *BaseNetworkController) GetLogicalPortName(pod *kapi.Pod, nadName string) string {
	if !bnc.IsSecondary() {
		return util.GetLogicalPortName(pod.Namespace, pod.Name)
	} else {
		return util.GetSecondaryNetworkLogicalPortName(pod.Namespace, pod.Name, nadName)
	}
}
```
and seems to do exactly the same thing, so I just suggest using it

Reply:
Ah, I see. I didn't even remember we have this function. thanks!